common/membership/resolver.go

```diff
@@ -84,6 +85,7 @@ type MultiringResolver struct {
 	status  int32
 
 	provider PeerProvider
+	mu       sync.Mutex
 	rings    map[string]*ring
```

Comment:
since this is the first instance of multiple accessors to the rings map, it needs guards